pandas/core/indexes/base.py

```diff
@@ -2947,7 +2951,8 @@ def difference(self, other):
         self._assert_can_do_setop(other)
 
         if self.equals(other):
-            return self._shallow_copy([])
+            # pass an empty np.ndarray with the appropriate dtype
+            return self._shallow_copy(self._data[:0])
```

Comment:
so we should really define an ``empty()`` on Index, Series, and EA.

Reply:
can you make an issue for this.

Reply:
will do.